internal/core/app_config.go

```diff
@@ -65,6 +65,9 @@ func (a *App) ConfigSync(ctx context.Context) error {
 			Project:          a.project.Ref(),
 			Name:             a.Ref().Application,
 			FileChangeSignal: appMD.FileChangeSignal,
+			RunnerProfile: &pb.Ref_OnDemandRunnerConfig{
+				Name: a.config.Runner.Profile,
```

Comment:
Oh, this panics when the `Runner` stanza isn't present in app config. Logs from a `waypoint up`:

```
...
✓ Docker image pushed: 797645259670.dkr.ecr.us-east-2.amazonaws.com/izaakprod:3334c2f2eb54120cf14190ec7ca0f3d18446de1e_CHANGES_1642009400

» Deploying backend...panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x40a47af]

goroutine 82 [running]:
github.com/hashicorp/waypoint/internal/core.(*App).ConfigSync(0xc000ea3ae0, 0x57576d8, 0xc000f1fa40, 0x0, 0x0)
	/Users/izaak/dev/waypoint/internal/core/app_config.go:69 +0x5ef
github.com/hashicorp/waypoint/internal/core.(*deployOperation).Do(0xc000edefc0, 0x57576d8, 0xc000f1fa40, 0x579fc98, 0xc0009b5080, 0xc000ea3ae0, 0x572dc10, 0xc00024b500, 0xc00024b500, 0x0, ...)
	/Users/izaak/dev/waypoint/internal/core/app_deploy.go:325 +0xa9
github.com/hashicorp/waypoint/internal/core.(*App).doOperation(0xc000ea3ae0, 0x57576d8, 0xc000f1fa40, 0x579fc98, 0xc0009b5080, 0x577ade8, 0xc000edefc0, 0x0, 0x0, 0x0, ...)
	/Users/izaak/dev/waypoint/internal/core/operation.go:136 +0x1107
github.com/hashicorp/waypoint/internal/core.(*App).Deploy(0xc000ea3ae0, 0x57576d8, 0xc000f1fa40, 0xc0015fe0c0, 0x0, 0x0, 0x0)
	/Users/izaak/dev/waypoint/internal/core/app_deploy.go:47 +0x27b
github.com/hashicorp/waypoint/internal/runner.(*Runner).executeDeployOp(0xc0011ba690, 0x57576d8, 0xc000f1fa40, 0xc0013ff3f0, 0xc0011a7740, 0x2, 0x0, 0x0)
	/Users/izaak/dev/waypoint/internal/runner/operation_deploy.go:26 +0xbd
github.com/hashicorp/waypoint/internal/runner.(*Runner).executeUpOp(0xc0011ba690, 0x57576d8, 0xc000f1fa40, 0x579fc98, 0xc000caea80, 0xc00004a780, 0xc0011a7740, 0x0, 0x0, 0x0)
	/Users/izaak/dev/waypoint/internal/runner/operation_up.go:55 +0x46b
github.com/hashicorp/waypoint/internal/runner.(*Runner).executeJob(0xc0011ba690, 0x57576d8, 0xc000f1fa40, 0x579fc98, 0xc000caea80, 0x5780ab8, 0xc0012fe798, 0xc00004a780, 0xc00007e0f4, 0x1d, ...)
	/Users/izaak/dev/waypoint/internal/runner/operation.go:278 +0x16b3
github.com/hashicorp/waypoint/internal/runner.(*Runner).prepareAndExecuteJob(0xc0011ba690, 0x57576d8, 0xc000f1fa40, 0x579fc98, 0xc000caea80, 0x5780ab8, 0xc0012fe798, 0xc000328328, 0x5780e28, 0xc00103f340, ...)
	/Users/izaak/dev/waypoint/internal/runner/accept.go:450 +0x731
github.com/hashicorp/waypoint/internal/runner.(*Runner).accept(0xc0011ba690, 0x57576d8, 0xc000f1fa40, 0x0, 0x0, 0x0, 0x0)
	/Users/izaak/dev/waypoint/internal/runner/accept.go:307 +0xee5
github.com/hashicorp/waypoint/internal/runner.(*Runner).Accept(...)
	/Users/izaak/dev/waypoint/internal/runner/accept.go:76
github.com/hashicorp/waypoint/internal/runner.(*Runner).AcceptMany(0xc0011ba690, 0x57576d8, 0xc0000cf900)
	/Users/izaak/dev/waypoint/internal/runner/accept.go:27 +0x4b
github.com/hashicorp/waypoint/internal/client.(*Project).startRunner.func1(0xc00102a7e0, 0xc0011ba690)
	/Users/izaak/dev/waypoint/internal/client/runner.go:47 +0x78
created by github.com/hashicorp/waypoint/internal/client.(*Project).startRunner
	/Users/izaak/dev/waypoint/internal/client/runner.go:45 +0x214
```

I think we can just nil-check `Runner` here, and wherever else we're looking at it. Or maybe initialize `Runner` to an empty struct.